client/components/breadcrumb/index.tsx

```diff
@@ -1,77 +1,164 @@
-	compact?: boolean;
+	items: Item[];
+	mobileItem: Item;
+	buttons?: JSX.Element[];
+	mobileButtons?: JSX.Element[];
 }
 
-const Breadcrumb: React.FunctionComponent< Props > = ( { items, compact = false } ) => {
-	const translate = useTranslate();
+const Breadcrumbs: React.FunctionComponent< Props > = ( {
+	items,
+	mobileItem,
+	buttons,
+	mobileButtons,
+} ) => {
+	const renderItemLabel = ( item: Item ) => {
+		if ( item.href ) {
+			return (
+				<a className="breadcrumb__item-label" href={ item.href }>
+					{ item.label }
+				</a>
+			);
+		}
+		return <span className="breadcrumb__item-label">{ item.label || item }</span>;
+	};
 
-	return (
-		<StyledUl>
-			{ compact && items.length > 1 ? (
-				<StyledLi>
-					<StyledGridicon icon="chevron-left" size={ 18 } />
-					{ /*  Show the exactly previous page with items[ items.length - 2 ] */ }
-					<a href={ items[ items.length - 2 ].href }>{ translate( 'Back' ) }</a>
-				</StyledLi>
-			) : (
-				<>
-					{ items.map( ( item: { href?: string; label: string }, index: Key ) => {
-						return (
-							<StyledLi key={ index }>
-								{ compact && <StyledGridicon icon="chevron-left" size={ 18 } /> }
-								{ index !== 0 && <StyledGridicon icon="chevron-right" size={ 18 } /> }
-								{ item.href ? (
-									<a href={ item.href }>{ item.label }</a>
-								) : (
-									<span>{ item.label }</span>
-								) }
-							</StyledLi>
-						);
-					} ) }
-				</>
+	const renderHelpBubble = ( item: Item ) => {
+		if ( ! item.helpBubble ) {
+			return null;
+		}
+		return (
+			<InfoPopover
+				className="breadcrumb__help-bubble"
+				icon="help-outline"
+				position={ 'right' }
+				screenReaderText={ 'Learn more' }
+			>
+				{ item.helpBubble }
+			</InfoPopover>
+		);
+	};
+
+	const renderSeparator = ( index: number ) => {
+		if ( index === items.length - 1 ) {
+			return null;
+		}
+		/* eslint-disable wpcalypso/jsx-gridicon-size */
+		return <Gridicon className="breadcrumb__separator" icon="chevron-right" size={ 14 } />;
+		/* eslint-enable wpcalypso/jsx-gridicon-size */
+	};
+
+	const renderItem = ( item: Item, index: number ) => {
+		const classes = classNames( 'breadcrumb__item', {
+			'is-last-item': index === items.length - 1,
+			'is-only-item': items.length === 1,
+		} );
+		return (
+			<React.Fragment key={ `breadcrumb${ index }` }>
+				<span className={ classes }>{ renderItemLabel( item ) }</span>
+				{ renderHelpBubble( item ) }
+				{ renderSeparator( index ) }
+			</React.Fragment>
+		);
+	};
+
+	const renderBackArrow = () => {
+		if ( mobileItem.showBackArrow && mobileItem.href ) {
+			/* eslint-disable wpcalypso/jsx-gridicon-size */
+			return (
+				<a className="breadcrumb__item" href={ mobileItem.href }>
+					<Icon className="breadcrumb__back-arrow" icon={ chevronLeft } size={ 20 } />
+				</a>
+			);
+			/* eslint-enable wpcalypso/jsx-gridicon-size */
+		}
+		return null;
+	};
+
+	const renderMobileItem = () => {
+		return (
+			<>
+				{ renderBackArrow() }
```

Comment:
```suggestion
				{ mobileItem.showBackArrow && mobileItem.href && (
					<a className="breadcrumb__item" href={ mobileItem.href }>
						<Icon className="breadcrumb__back-arrow" icon={ chevronLeft } size={ 20 } />
					</a>
				) }
```

I wonder if inlining would help with readability and avoid us using extraneous functions.